src/main/java/com/example/Library/controller/AuthenticationRestController.java

```diff
@@ -26,22 +26,22 @@
 @RequestMapping("/api/v1/auth")
 public class AuthenticationRestController {
 
-    public AuthenticationRestController(AuthenticationManager authenticationManager, UserRepository userRepository, JwtTokenProvider jwtTokenProvider) {
+    public AuthenticationRestController(AuthenticationManager authenticationManager, UserService userService, JwtTokenProvider jwtTokenProvider) {
         this.authenticationManager = authenticationManager;
-        this.userRepository = userRepository;
+        this.userService = userService;
         this.jwtTokenProvider = jwtTokenProvider;
     }
 
     private final AuthenticationManager authenticationManager;
-    private UserRepository userRepository;
+    private UserService userService;
     private JwtTokenProvider jwtTokenProvider;
 
     @PostMapping("/login")
     public ResponseEntity<?> authenticate(@RequestBody AuthenticationRequestDTO request){
```

Comment:
authentication logic could be extracted from the controller to new AuthenticationService with 2 methods: 
1) authenticate
2) logout